naming/src/main/java/org/wildfly/httpclient/naming/HttpRemoteNamingService.java

```diff
@@ -101,21 +101,34 @@ private void registerHandler(final RoutingHandler routingHandler, final RequestT
 
     private HttpHandler newInvocationHandler(final RequestType requestType) {
         switch (requestType) {
-            case BIND: return new BindHandler();
-            case CREATE_SUBCONTEXT: return new CreateSubContextHandler();
-            case DESTROY_SUBCONTEXT: return new DestroySubcontextHandler();
-            case LIST: return new ListHandler();
-            case LIST_BINDINGS: return new ListBindingsHandler();
-            case LOOKUP: return new LookupHandler();
-            case LOOKUP_LINK: return new LookupLinkHandler();
-            case REBIND: return new RebindHandler();
-            case RENAME: return new RenameHandler();
-            case UNBIND: return new UnbindHandler();
+            case BIND: return new BindHandler(localContext, httpServiceConfig, classResolverFilter);
```

Comment:
What is the motivation for making these static?  I can see you have also parameterized the constructors for eventual use in the ServerHandlers factory.

Reply:
I made them static to be independent of HttpRemoteNamingService class - preparation step to extracting server handlers out from this class @rachmatowicz .